src/CSFML/Network/Ftp.cpp

```diff
@@ -49,11 +49,11 @@ static_assert(alignof(sfChar32) == alignof(char32_t));
 ////////////////////////////////////////////////////////////
 [[nodiscard]] sfChar32* copyToChar32(const sf::String& str)
 {
-    const std::size_t byteCount = sizeof(sfChar32) * str.getSize();
-    auto*             utf32     = static_cast<sfChar32*>(std::malloc(byteCount + sizeof(sfChar32)));
+    const std::size_t byteCount = sizeof(sfChar32) * (str.getSize() + 1);
+    auto*             utf32     = static_cast<sfChar32*>(std::malloc(byteCount));
+    if (!utf32)
+        return nullptr;
     std::memcpy(utf32, str.getData(), byteCount);
-    utf32[str.getSize()] = 0;
-
     return utf32;
```

Comment:
Can you talk me through why `byteCount` changed and why `utf32[str.getSize()] = 0;` was removed? The commit message only talks about checking a return value so I was not expecting to see deeper changes to the implementation.

Reply:
I didn't like the repeating of the typename sfChar32 twice (once in malloc call and once in byteCount) and I used the fact that C++11 strings are now always null terminated in the memcpy call and can now remove explicit setting of null terminator.

should have mentioned it in the pr message.